sklearn/linear_model/base.py

```diff
@@ -273,6 +273,11 @@ def _set_intercept(self, X_offset, y_offset, X_scale):
         """Set the intercept_
         """
         if self.fit_intercept:
+            if isinstance(self.coef_, np.ndarray):
```

Comment:
This can be done away with, I believe. Look at `_preprocess_data` and see for places where it is possible that `X_offset` can be explicitly can be `float64`